features/ui/input/input.tsx

```diff
@@ -0,0 +1,145 @@
+import { color, textFont } from "@styles/theme";
+import { InputHTMLAttributes, useRef } from "react";
+import styled from "styled-components";
+import { nanoid } from "nanoid";
+
+export type InputProps = InputHTMLAttributes<HTMLInputElement> & {
+  label?: string | undefined;
+  iconSrc?: string | undefined;
+  error?: boolean;
+  errorMsg?: string | undefined;
+  hintMsg?: string | undefined;
+};
+
+const Container = styled.label`
+  display: flex;
+  flex-direction: column;
+  max-width: 20rem;
+`;
+
+const LabelText = styled.span`
+  ${textFont("sm", "medium")}
+  color: ${color("gray", 700)};
+  padding-bottom: 6px;
+`;
+
+const InputContainer = styled.div`
+  position: relative;
+`;
+
+const StyledInput = styled.input<{
+  error: boolean;
+  iconSrc: string | undefined;
+}>`
+  box-sizing: border-box;
+  width: 100%;
+  padding-top: 9px;
+  padding-bottom: 9px;
+  padding-left: ${(props) => (props.iconSrc ? "42px" : "14px")};
+  padding-right: ${(props) => (props.error ? "38px" : "14px")};
+  background: #ffffff;
+  border: 1px solid
+    ${(props) => (props.error ? color("error", 300) : color("gray", 300))};
+  box-shadow: 0px 1px 2px rgba(16, 24, 40, 0.05);
+  border-radius: 8px;
+  ${textFont("md", "regular")};
+  color: ${color("gray", 900)};
+
+  &::placeholder {
+    color: ${color("gray", 500)};
+    ${textFont("md", "regular")};
+  }
+
+  &:focus {
+    border-color: ${(props) =>
+      props.error ? color("error", 300) : color("primary", 300)};
+    box-shadow: 0px 1px 2px rgba(16, 24, 40, 0.05),
+      0px 0px 0px 4px ${(props) => (props.error ? "#FEE4E2" : "#F4EBFF")};
```

Comment:
You can use the error colors from the theme here. E.g. `color(props.theme)("error", 100)`. Not sure if that's the right color though :) Alternativley you can also use the ternary only to decide on the shade (second parameter of `color(props.theme)(...)`